src/ThemePage.tsx

```diff
@@ -1,46 +1,50 @@
-import { useParams, useNavigate } from "react-router";
+import { useParams } from "react-router";
 import PageContainer from "@/components/PageContainer";
-import { useEffect, useState } from "react";
-import { getThemeInfo, type ThemeID, type ThemeInfo } from "@/apis/theme";
-import { AxiosError, HttpStatusCode } from "axios";
 import ThemeHeroSection from "@/sections/ThemeSection/ThemeHeroSection";
 import ThemeProductSection from "@/sections/ThemeSection/ThemeProductSection"
+import { useThemeInfoQuery } from "./hooks/useThemeInfoQuery";
+import { Suspense } from "react";
+import { ErrorBoundary } from "react-error-boundary";
+import { QueryErrorResetBoundary } from "@tanstack/react-query";
+import Spinner from "./components/Spinner";
 
-export default function ThemePage() {
-    const { themeId } = useParams<{ themeId: ThemeID}>();
-    const navigate = useNavigate();
-    const [themeInfo, setThemeInfo] = useState<ThemeInfo | null>(null);
-
-    useEffect(() => {
-        if (!themeId) {
-            navigate("/");
-            return;
-        }
+function ThemeInfoSection({ themeId }: { themeId: string }) {
+    const { data } = useThemeInfoQuery(themeId);
+    return (
+        <ThemeHeroSection
+            name={data.name}
+            title={data.title}
+            description={data.description}
+            backgroundColor={data.backgroundColor}
+        />
```

Comment:
```suggestion
        <ThemeHeroSection {...data} />
```
key 값이 동일하면 이렇게 spread 문법을 사용할 수 있습니다.